internal/provider/networking_resource.go

```diff
@@ -126,8 +126,8 @@ func (n allowListResource) Create(ctx context.Context, req tfsdk.CreateResourceR
 
 	if cluster.Config.Serverless != nil {
 		resp.Diagnostics.AddError(
-			"Could not add network allow list in serverless cluster",
-			fmt.Sprintf("Network allow list is a feature of dedicated cluster: %s", formatAPIErrorMessage(err)),
+			"Could not add network allowlist in serverless cluster",
+			"Network allowlists are only supported with dedicated clusters",
```

Comment:
It's the presence of a serverless config, not lack. This is a validation step, and in a future version, this will happen at plan time. That plan-time validation (and corresponding dry run) will happen way before we add these sorts of network tools to serverless clusters.